File: Python3/프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.py
def solution(num, total):
    answer = []
    
    start_num, middle_index = total // num, 0
    if num % 2 == 0:
        middle_index = num // 2 - 1
    else:
        middle_index = num // 2
    
    for i in range(num):
        answer.append(start_num - middle_index + i)
    
    return answer    

File: Swift/프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.swift
import Foundation

func solution(_ num:Int, _ total:Int) -> [Int] {
    var answer: [Int] = []
    
    let startNum = Int(total / num)
    var middleIndex = 0
    
    if num % 2 == 0 {
        middleIndex = Int(num / 2) - 1
    } else {
        middleIndex = Int(num / 2)
    }
    
    for i in 0..<num {
        answer.append(startNum - middleIndex + i)
    }
    
    return answer
}